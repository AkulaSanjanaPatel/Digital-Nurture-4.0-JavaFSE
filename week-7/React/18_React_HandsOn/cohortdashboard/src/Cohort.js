export const CohortData = [
  {
    code: 'COH2025',
    name: 'Full Stack Development',
    status: 'Ongoing'
  },
  {
    code: 'COH2024',
    name: 'Data Science',
    status: 'Completed'
  }
];
