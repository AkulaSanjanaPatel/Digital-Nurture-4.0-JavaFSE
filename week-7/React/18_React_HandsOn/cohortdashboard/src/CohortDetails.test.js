import React from 'react';
import { render, screen } from '@testing-library/react';
import CohortDetails from './CohortDetails';
import { CohortData } from './Cohort';

describe('CohortDetails Component', () => {
  test('should render the component', () => {
    render(<CohortDetails cohort={CohortData[0]} />);
    expect(screen.getByText('COH2025')).toBeInTheDocument();
  });

  test('should display cohort name and status', () => {
    render(<CohortDetails cohort={CohortData[0]} />);
    expect(screen.getByText('Name: Full Stack Development')).toBeInTheDocument();
    expect(screen.getByText('Status: Ongoing')).toBeInTheDocument();
  });

  test('should match snapshot', () => {
    const { asFragment } = render(<CohortDetails cohort={CohortData[0]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
