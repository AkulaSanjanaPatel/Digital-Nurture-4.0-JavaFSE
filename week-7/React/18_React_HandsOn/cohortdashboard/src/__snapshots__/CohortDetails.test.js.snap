// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CohortDetails Component should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      COH2025
    </h3>
    <p>
      <strong>
        Name:
      </strong>
       Full Stack Development
    </p>
    <p>
      <strong>
        Status:
      </strong>
       Ongoing
    </p>
  </div>
</DocumentFragment>
`;
